impeller/entity/contents/filters/gaussian_blur_filter_contents_unittests.cc

```diff
@@ -0,0 +1,242 @@
+      //                  expand the coverage. See note inside of
+      //                  gaussian_blur_filter_contents.cc.
+      // EXPECT_TRUE(RectNear(result_coverage.value(),
+      //                      contents_coverage.value()));
+      EXPECT_TRUE(RectNear(contents_coverage.value(),
+                           Rect::MakeLTRB(99, 199, 201, 301)));
+      EXPECT_TRUE(RectNear(result_coverage.value(),
+                           Rect::MakeLTRB(100, 200, 200, 300)));
+    }
+  }
+}
+
+TEST_P(GaussianBlurFilterContentsTest,
+       RenderCoverageMatchesGetCoverageRotated) {
+  TextureDescriptor desc = {
+      .format = PixelFormat::kB8G8R8A8UNormInt,
+      .size = ISize(400, 300),
+  };
+  std::shared_ptr<Texture> texture = MakeTexture(desc);
+  Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
+  auto contents = std::make_unique<GaussianBlurFilterContents>(sigma_radius_1);
+  contents->SetInputs({FilterInput::Make(texture)});
+  std::shared_ptr<ContentContext> renderer = GetContentContext();
+
+  Entity entity;
+  entity.SetTransformation(Matrix::MakeTranslation({400, 100, 0}) *
+                           Matrix::MakeRotationZ(Degrees(90.0)));
+  std::optional<Entity> result =
+      contents->GetEntity(*renderer, entity, /*coverage_hint=*/{});
+  EXPECT_TRUE(result.has_value());
+  if (result.has_value()) {
+    EXPECT_EQ(result.value().GetBlendMode(), BlendMode::kSourceOver);
+    std::optional<Rect> result_coverage = result.value().GetCoverage();
+    std::optional<Rect> contents_coverage = contents->GetCoverage(entity);
+    EXPECT_TRUE(result_coverage.has_value());
+    EXPECT_TRUE(contents_coverage.has_value());
+    if (result_coverage.has_value() && contents_coverage.has_value()) {
+      // TODO(gaaclarke): This test won't pass until the blur_radius is used to
+      //                  expand the coverage. See note inside of
+      //                  gaussian_blur_filter_contents.cc.
+      // EXPECT_TRUE(RectNear(result_coverage.value(),
+      //                      contents_coverage.value()));
+      EXPECT_TRUE(RectNear(contents_coverage.value(),
+                           Rect::MakeLTRB(99, 99, 401, 501)));
+      EXPECT_TRUE(RectNear(result_coverage.value(),
+                           Rect::MakeLTRB(100, 100, 400, 500)));
+    }
+  }
+}
+
+TEST_P(GaussianBlurFilterContentsTest, CalculateUVsSimple) {
+  TextureDescriptor desc = {
+      .format = PixelFormat::kB8G8R8A8UNormInt,
+      .size = ISize(100, 100),
+  };
+  std::shared_ptr<Texture> texture = MakeTexture(desc);
+  auto filter_input = FilterInput::Make(texture);
+  Entity entity;
+  Quad uvs = GaussianBlurFilterContents::CalculateUVs(filter_input, entity,
+                                                      ISize(100, 100));
+  std::optional<Rect> uvs_bounds = Rect::MakePointBounds(uvs);
+  EXPECT_TRUE(uvs_bounds.has_value());
+  if (uvs_bounds.has_value()) {
+    EXPECT_TRUE(RectNear(uvs_bounds.value(), Rect::MakeXYWH(0, 0, 1, 1)));
+  }
+}
```

Comment:
If we used mock objects as we had originally I could test the uv values directly when rendering.  This is the best I can do with real objects.  cc @matanlurey @jonahwilliams

Reply:
In theory you could instrument the creation of commands and render passes, and then read-back the real UVs. I might implement something like that in the future.

In the meantime I think its fair to say that based on the gaussian structure it would be too difficult to pull out the real UVs (by hooking into the cmd buffer) and that mocks may be appropriate. Factoring out the logic into a testable method is also a completely reasonable means to test IMO.

But @gaaclarke you're the one putting in the work here. If you don't feel like the static methods are sufficient then mock away.

Reply:
Yea, the issue I ran into is that I kept fixing one test and breaking another and golden image tests cannot be run locally so I have to do manual testing to make sure those 2 tests kept working.  Ideally I'd want to scale beyond checking 2 tests too.  I'll see if I get around to testing uvs directly via mocks.